lua/openingh/utils.lua

```diff
@@ -77,17 +77,21 @@ end
 -- Returns the current branch or commit if they are available on remote
 -- otherwise this will return the default branch of the repo
 function M.get_current_branch_or_commit()
-  local current_branch = get_current_branch()
-  if current_branch ~= "HEAD" and M.is_branch_upstreamed(current_branch) then
-    return current_branch
+  local core = function()
+    local current_branch = get_current_branch()
+    if current_branch ~= "HEAD" and M.is_branch_upstreamed(current_branch) then
+      return current_branch
+    end
+
+    local commit_hash = get_current_commit_hash()
+    if current_branch == "HEAD" and M.is_commit_upstreamed(commit_hash) then
+      return commit_hash
+    end
+
+    return M.get_default_branch()
   end
 
-  local commit_hash = get_current_commit_hash()
-  if current_branch == "HEAD" and M.is_commit_upstreamed(commit_hash) then
-    return commit_hash
-  end
-
-  return M.get_default_branch()
+  return string.gsub(core(), "#", "%%23")
```

Comment:
Why don't we encode the whole url instead of replacing the #char

Reply:
To be honest converting only `#` is the easiest way for me to avoid the problem I faced.

And I'm not confident about what should be and should not be encoded for github.
I thought that the character `/` should not be converted, though...
I then, now, confirmed that both `/` and `%2F` are okay for branch.
(But it must not be okay for whole file path)

In addition, I didn't know that it's also okay to convert normal ascii characters `a-z`, `0-9` and so on.

Then, something like following could be an another solution not only limited to `#` but all the other characters, you mean?

```lua
string.gsub(branch_or_commit_hash, ".", function (c) return string.format("%%%02X", string.byte(c)) end))
```

Reply:
I locally created a commit that does another solution and applied it to file path.
What do you think?
https://github.com/msr1k/openingh.nvim/compare/374c081409dc238018cd986410f16190d8f9f680..d0a3eac853acfe213461bb9921a6b0f4632bbbf1